tidy3d/components/source.py

```diff
@@ -673,6 +673,14 @@ class TFSF(AngledFieldSource, VolumeSource):
 
 
 # sources allowed in Simulation.sources
-SourceType = Union[
-    UniformCurrentSource, PointDipole, GaussianBeam, AstigmaticGaussianBeam, ModeSource, PlaneWave
+SourceType = Annotated[
+    Union[
+        UniformCurrentSource,
+        PointDipole,
+        GaussianBeam,
+        AstigmaticGaussianBeam,
+        ModeSource,
+        PlaneWave,
+    ],
+    pydantic.Field(discriminator="type"),
```

Comment:
The string "type" is set by a variable called ``TYPE_TAG_STR`` in `tidy3d/components/base.py`. 

So let's import it

```python
from tidy3d.components.base import TYPE_TAG_STR
```

and use its value in the discriminator
```py
discriminator=TYPE_TAG_STR
```

Reply:
> This should be ok. note that it may affect the schema to add the annotation. Did you ever find out what the pydantic version is? We don't see this error on our side running pydantic 1.9.1.

I use the pydantic 1.9.1 and also get the issue.  Sample code is here [tidy3d issue](https://gist.github.com/JunchengXue/f7a94ac8e3932e0017a002e2ed364af5)

Reply:
If you use python code to generate Simulation, the thrown error is correct, it only happens during the process of parsing json to python object